python/smashbox/config.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os.path
+import pickle
+from pathlib import Path
+from typing import Any, Literal
+
+import yaml
+from pydantic import BaseSettings
+
+
+logger = None
+
+# this should probably be moved into a utilities module
+def get_logger(name: str = "config", level: int | None = None) -> logging.Logger:
+    global logger
+    if not logger:
+      if level is None:
+          level = logging.INFO  # change here to DEBUG if you want to debug config stuff
+      logging.basicConfig(level=level)
+    return logging.getLogger('.'.join(['smash', name]))
+
+
+class Configuration(BaseSettings):
+    """Root configuration object that parses the values defined in the config file."""
+    smashdir: str
+    oc_account_name: str | None
+    oc_number_test_users: int
+    oc_group_name: str | None
+    oc_number_test_groups: int
+    oc_account_password: str
+    oc_server: str
+    oc_root: str
+    oc_server_folder: str
+    oc_ssl_enabled: bool
+    oc_server_shell_cmd: str
+    oc_server_tools_path: str  # TODO: is this still needed?
+    oc_sync_cmd: str
+    oc_sync_repeat: int
+
+    runid: int | None
+    workdir_runid_enabled: bool
+    oc_account_runid_enabled: bool
+
+    oc_account_reset_procedure: Literal["delete", "keep"]  # there are some more types that are not yet implemented
+    rundir_reset_procedure: Literal["delete", "keep"]
+
+    web_user: str
+    oc_admin_user: str
+    oc_admin_password: str
+
+    scp_port: int
+    oc_server_log_user: str
+    oc_check_server_log: bool
+
+    @property
+    def oc_webdav_endpoint(self) -> str:
+        return os.path.join(self.oc_root, "remote.php/webdav")
+
+    @property
+    def oc_server_datadirectory(self) -> str:
+        return os.path.join("/var/www/html", self.oc_root, "data")
+    
+    # these methods exists for backwards compatibility
```

Comment:
Do you intend to refactor those in the future? As demonstrated correctly below, Python's `getattr` and `dict` facilities are perfect Pythonic replacements.